mod/src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/ScoreCalculation.kt

```diff
@@ -260,55 +346,100 @@ object ScoreCalculation {
-                ScoreCalculationElement.text.add("§f• §eFailed Puzzles:§c ${failedPuzzles.get()}")
-                if (foundSecrets.get() > 0) ScoreCalculationElement.text.add(
-                    "§f• §eSecrets: ${if (foundSecrets.get() >= totalSecretsNeeded.get()) "§a" else "§c"}${foundSecrets.get()}§7/§a${totalSecretsNeeded.get()} " +
-                            if (floorReq.get().secretPercentage != 1.0) "§7(§6Total: ${totalSecrets.get()}§7)" else ""
-                )
-                ScoreCalculationElement.text.add("§f• §eCrypts:§a ${crypts.get()}")
-                if (dungeonFloorNumber?.let { it >= 6 } == true) {
-                    ScoreCalculationElement.text.add("§f• §eMimic:§l${if (mimicKilled.get()) "§a ✔" else "§c ✘"}")
-                }
+                ScoreCalculationElement.text.add("• §eDeaths: ${if (deaths.get() == 0) {
+                    "§a"
+                } else {
+                    "§c"
+                }}${deaths.get()} ${if (firstDeathHadSpirit.get()) "§7(§6Spirit§7)" else ""}")
+                ScoreCalculationElement.text.add("• §ePuzzles: ${if (completedPuzzleCount == puzzleCount) {
+                    "§a"
+                } else if (failedPuzzles.get() > 0) {
+                    "§c"
+                } else {
+                    "§e"
+                }}$completedPuzzleCount§7/§a$puzzleCount")
+                ScoreCalculationElement.text.add("• §eSecrets: " + if (totalSecrets.get() == 0) {
+                    "§7?"
+                } else {
+                    val found = foundSecrets.get()
+                    (if (found == totalSecrets.get()) {
+                        "§a"
+                    } else if (found >= minSecrets.get()) {
+                        "§e"
+                    } else {
+                        "§c"
+                    }) + found + "§7/§a" + totalSecrets.get() + " §e(min: " + when (val sec = minSecrets.get()) {
+                        -1 -> "§7?"
+                        -2 -> "§c✘"
+                        -3 -> "§a✔"
+                        else -> "§a${(sec - found).takeUnless { it <= 0 } ?: "✔"}"
+                    } + "§e, could skip: " + when (val skip = skippableSecrets.get()) {
+                        -1 -> "§7?"
+                        -2 -> "§c✘"
+                        -3 -> "§a✔"
+                        else -> "§a${totalSecrets.get()  - skip}"
+                    } + "§e)"
+                })
+                ScoreCalculationElement.text.add("• §eCrypts: ${if (crypts.get() >= 5) {
+                    "§a"
+                } else {
+                    "§c"
+                }}${crypts.get().coerceAtMost(5)}")
+                ScoreCalculationElement.text.add("• §eMimic: §l${if ((dungeonFloorNumber ?: 0) >= 6) {
+                    if (mimicKilled.get()) {
+                        "§a✔"
+                    } else {
+                        "§c✘"
+                    }
+                } else {
+                    "§7-"
+                }}")
```

Comment:
Perhaps it would make more sense to expand everything out into multiline string templates to avoid the awkward formatting here. In particular, the if/else statements being expanded while the rest of the string template is not feels weird and also the prevalence of using the same colors for a simple boolean which could maybe be replaced with a helper method.

Reply:
Kinda find it awkward to have only half of the lines on the template, and moving puzzles and secrets outside of it, but it did improve readability a lot, probably worth settling with [it](https://i.imgur.com/iMTbu6A.png) either way?

Reply:
Yeah that looks better